crates/air-lib/test-utils/src/test_runner.rs

```diff
@@ -39,20 +41,40 @@ pub trait AirRunner {
         ttl: u32,
         override_current_peer_id: Option<String>,
         call_results: avm_server::CallResults,
+        key_pair: &KeyPair,
     ) -> Result<RawAVMOutcome, Box<dyn std::error::Error>>;
 }
 
 pub struct TestRunner<R = AirRunnerImpl> {
     pub runner: R,
-    pub call_service: CallServiceClosure,
+    call_service: CallServiceClosure,
+    key_secret: KeyPair,
 }
 
-#[derive(Debug, Default, Clone)]
+#[derive(Default, Clone)]
 pub struct TestRunParameters {
     pub init_peer_id: String,
     pub timestamp: u64,
     pub ttl: u32,
-    pub override_current_peer_id: Option<String>,
+    pub override_current_peer_id: Option<(String, KeyPair)>,
```

Comment:
wow, why use tuple and not add a new field for key pair?

field is called `..._peer_id`, but contains String and KeyPair. That makes me dizzy, code below becomes pretty hard to understand, eg
```
        let key_pair = override_current_peer_id
             .as_ref()
             .map(|pair| &pair.1)
             .unwrap_or(&self.key_secret);
         let override_current_peer_id = override_current_peer_id
             .clone()
             .map(|(peer_id, _key)| peer_id);
```